src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/FluentWaitCondition.java

```diff
@@ -1,36 +1,35 @@
 package com.wiley.autotest.selenium.elements.upgrade.v3;
 
+import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.support.ui.FluentWait;
 
 import java.util.function.Function;
 
 /**
  * Created by vefimov on 18/04/2017.
  */
-public class FluentWaitCondition<T> extends FluentWait<T> {
+public class FluentWaitCondition extends FluentWait<WebDriver> {
 
     //default condition should stop execution in case of failure
     private boolean nullOnFailure = false;
 
-    public FluentWaitCondition(T input) {
+    public FluentWaitCondition(WebDriver input) {
         super(input);
     }
 
     void setNullOnFailure(boolean nullOnFailure) {
         this.nullOnFailure = nullOnFailure;
     }
 
-    void waitFor(Function<T, Boolean> condition) {
-        if (nullOnFailure) {
-            try {
-                until(condition);
-            } catch (Throwable ignoredAndContinue) {
-                //todo add some logging here
-            }
-        } else {
+    <T> void waitFor(Function<WebDriver, T> condition) {
+        try {
             until(condition);
+        } catch (Throwable ignoredAndContinue) {
+            if (nullOnFailure) {
+                //todo add some logging here if necessary
+            } else {
+                throw new AssertionError("Unable to perform: " + condition.toString());
```

Comment:
let's rephrase to something like "Condition: ***'" + condition.toString() "'*** failed!"